packages/zosfiles/src/cli/hRecall/ds/Ds.handler.ts

```diff
@@ -0,0 +1,28 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { AbstractSession, IHandlerParameters } from "@zowe/imperative";
+import { IZosFilesResponse } from "../../../api/doc/IZosFilesResponse";
+import { ZosFilesBaseHandler } from "../../ZosFilesBase.handler";
+import { HRecall, IRecallOptions } from "../../../api";
+
+/**
+ * Handler to rename a Data Set
+ */
+export default class DSHandler extends ZosFilesBaseHandler {
+  public async processWithSession(
+    commandParameters: IHandlerParameters,
+    session: AbstractSession,
+    options: Partial<IRecallOptions>): Promise<IZosFilesResponse> {
+
+    return HRecall.dataSet(session, commandParameters.arguments.dataSetName, options);
```

Comment:
The 3rd object provided by the `processWithSession` is referring to the zosmf profile and not the `commandParammeters.arguments`. 
This is then passed to the APIs which are expecting an `IRecallOptions` type of object with a `wait` property in order to add the corresponding header to the payload.

The default behavior of the `X-IBM-Migrated-Recall` is to always wait for the request.
This applies to hmigrate, hrecall, hdelete, and more.

Reply:
Hey @zFernand0, so for this you would like me to remove the 3rd object being `options` and make it a part of the `commandParameters.arguement`?

Reply:
Yeah, I believe that'd fix it 😋